### ISSUE TITLE
Combine all children into one device

### ISSUE THREAD

**Issue by flatsiedatsie** [tsathishkumar/MySController-rs#24]
Let's say I have a weather station that outputs
- temperature
- humidity
- barometer (air pressure).

Currently, the children of these devices are all presented as separate things in the Mozilla Gateway interface.

In theory the mozilla interface supports complex combined devices:
http://linuxgizmos.com/files/mozilla_things_light.jpg

Is it possible / is it a goal to also allow for this combination?

**Comment by flatsiedatsie** [comment]
A hacky way around this is to place things together on the floorplan view.

**Comment by freaktechnik** [comment]
If you won't mind I'll reply here to the questions in https://github.com/mozilla-iot/gateway/issues/1191#issuecomment-440251554

> Do you perhaps have a picture of what this multi-property output looks like?

![image](https://user-images.githubusercontent.com/640949/48773844-da025380-ecc8-11e8-99d4-0af7f20667d4.png)

or https://github.com/mozilla-iot/gateway/issues/1129#issuecomment-400044433 for an extreme example.

I don't have any device where there are multiple properties of exactly the same type, but I do know that there are some smartplugs with that characteristic, since they may have multiple outlets and all outlets have the same controls. This directly leads into:

> Dust sensor with 3 children, each indicating the total weight of dust particles detected of three different sizes. Optionally this could also send 3 "raw" particle counts.

This would probably be best solved with the object property nesting that the gateway doesn't support.

> Weather station, as mentioned, with 4 children: temp, hum, baro and weather prediction (text).

![image](https://user-images.githubusercontent.com/640949/48773859-e38bbb80-ecc8-11e8-9c17-9edfb2ae4ffd.png)

> A smart lock with 9 children.

You may be interested in https://github.com/mozilla-iot/schemas/issues/11

Lastly, also note that you can provide a URL to a custom webpage for the state and control of a thing. (https://mozilla-iot.github.io/wot/#links-member `rel: alternate`)

**Comment by freaktechnik** [comment]
Ah, that is not possible. It could in theory be possible with a matching capability, but that's not something the UI currently tries to do. The value shown in the overview is intended to give you a rough idea of the state of your things, and not detail exactly what state they're in. One example of showing multiple values is colored lights, those will reflect both their current color/brightness/hue and their on/off state in the preview.

**Comment by flatsiedatsie** [comment]
If that's not possible, well, that's something for Mozilla to explore I think. All other controller allow this in some form.

I quite like the [HomeGenie vibe](https://www.mysensors.org/uploads/580ca465176d6eb75165eb0e/394/hgweb_01.jpeg).

**Comment by flatsiedatsie** [comment]
Let's shelve this issue until Mozilla fixes this somehow?

**Comment by tsathishkumar** [comment]
Yes, that makes sense. I'll also need to check the webthing-rust crate if this is something they are planning to work on.

**Comment by tsathishkumar** [comment]
@flatsiedatsie I think there also need a change in MySController to create child-sensors as `properties` instead of `things`. I'm thinking of making `node` as the `thing` and all the child sensors as `properties` of the thing. Does that make sense?

**Comment by flatsiedatsie** [comment]
Still, I think I would also prefer doing it the official way at this point.. This is what my thing overview looks like currently:

![all_things](https://user-images.githubusercontent.com/805405/49206113-f2e5c580-f3a8-11e8-97ca-8497a6bfa78f.png)

**Comment by tsathishkumar** [comment]
May be as a starter, we can try making the child sensors as properties and see how that looks like. From there we can take it forward. I'll create a branch for the change and let you know once it is in working condition.

**Comment by flatsiedatsie** [comment]
Just curious: is there any progress on this?

**Comment by tsathishkumar** [comment]
Sorry, I didn't work on it for sometime now. I'll try to make sometime in next week and look into it.

**Comment by flatsiedatsie** [comment]
Did you get a chance to explore? Is there anyway I can help?

**Comment by tsathishkumar** [comment]
No progress yet. I got busy in my daily job. I will make sometime next week and work on this.

**Comment by flatsiedatsie** [comment]
I'm working on an alternative solution, so don't worry if you don't have the time.
